model/project_ref.go

```diff
@@ -1255,6 +1304,180 @@ func (projectRef *ProjectRef) Upsert() error {
 	return err
 }
 
+// DefaultSectionToRepo modifies a subset of the project ref to use the repo values instead.
+// This subset is based on the pages used in Spruce.
+func (p *ProjectRef) DefaultSectionToRepo(section ProjectRefSection, caller string) error {
+	before, err := GetProjectSettingsEvents(p)
+	if err != nil {
+		return errors.Wrap(err, "error getting before project settings event")
+	}
+
+	switch section {
+	// check if the sections cover all the settings
+	case ProjectRefGeneralSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					ProjectRefEnabledKey:                 1,
+					ProjectRefBatchTimeKey:               1,
+					ProjectRefRemotePathKey:              1,
+					projectRefSpawnHostScriptPathKey:     1,
+					projectRefDispatchingDisabledKey:     1,
+					ProjectRefDeactivatePreviousKey:      1,
+					projectRefRepotrackerDisabledKey:     1,
+					projectRefDefaultLoggerKey:           1,
+					projectRefCedarTestResultsEnabledKey: 1,
+					projectRefPatchingDisabledKey:        1,
+					projectRefTaskSyncKey:                1, // TODO: look into whether this works for individual settings
+					ProjectRefDisabledStatsCacheKey:      1,
+					ProjectRefFilesIgnoredFromCacheKey:   1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefAccessSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					ProjectRefPrivateKey:    1,
+					ProjectRefRestrictedKey: 1,
+					ProjectRefAdminsKey:     1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefVariablesSection:
+		err = db.Update(ProjectVarsCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					projectVarsMapKey:    1,
+					privateVarsMapKey:    1,
+					restrictedVarsMapKey: 1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefGithubSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
```

Comment:
Github webhooks isn't a setting that the user can modify; if the owner/repo/branch is reachable by Evergreen then it's true, otherwise it's false.
Github patch definitions and git tag version definitions are handled by the alias section below, i.e. we remove all aliases relating to Evergreen processes.

Reply:
Should they maybe be part of this section since it's part of the github page, and the user would expect that do be defaulted as well when the page is defaulted?

Reply:
moving this conversation to a different thread and resolving this one.